pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -10963,39 +10110,39 @@ spec:
           name: hello-world
 `)}
 
-			ts := []*v1beta1.Task{simpleHelloWorldTask}
+	ts := []*v1beta1.Task{simpleHelloWorldTask}
 
-			cms := []*corev1.ConfigMap{withEmbeddedStatus(newFeatureFlagsConfigMap(), tc.embeddedStatusVal)}
+	cms := []*corev1.ConfigMap{newFeatureFlagsConfigMap()}
 
-			d := test.Data{
-				PipelineRuns: prs,
-				Tasks:        ts,
-				ConfigMaps:   cms,
-			}
-			prt := newPipelineRunTest(t, d)
-			defer prt.Cancel()
+	d := test.Data{
+		PipelineRuns: prs,
+		Tasks:        ts,
+		ConfigMaps:   cms,
+	}
+	prt := newPipelineRunTest(t, d)
+	defer prt.Cancel()
 
-			pr, clients := prt.reconcileRun("foo", pipelineRunName, []string{}, false)
+	pr, clients := prt.reconcileRun("foo", pipelineRunName, []string{}, false)
 
-			if tc.embeddedStatusVal != config.MinimalEmbeddedStatus {
-				if len(pr.Status.Runs) > 1 {
-					t.Errorf("Expected one CustomRun in status, but found %d", len(pr.Status.Runs))
-				}
-				if len(pr.Status.TaskRuns) > 0 {
-					t.Errorf("Expected no TaskRuns in status, but found %d", len(pr.Status.TaskRuns))
-				}
-			}
-			if tc.embeddedStatusVal != config.FullEmbeddedStatus {
-				if len(pr.Status.ChildReferences) > 1 {
-					t.Errorf("Expected one CustomRun or TaskRun in child references, but found %d", len(pr.Status.ChildReferences))
-				}
-			}
+	for _, childRef := range pr.Status.ChildReferences {
+		var runsCount, taskrunsCount int
+		switch childRef.Kind {
+		case "TaskRun":
+			taskrunsCount++
+		case "Run":
+			runsCount++
+		}
+		if runsCount > 1 {
```

Comment:
TODO: figure out if this is the intended cases or I shall change to case "CustomRun"

Reply:
Thanks for checking @XinruZhang